### PR TITLE
Autobahn changes

### PR DESCRIPTION
**Pull request by marchinram**

### CODE REVIEW

File: Sources/CloseCode.swift [hunk 1]
```diff
@@ -39,7 +39,7 @@ public enum CloseCode {
     case TLSHandshake
     case Raw(UInt16)
```
Comment (paulofaria) [inline Sources/CloseCode.swift]: New API design guidelines states that the `cases` should ne lowerCamelCase.

``` swift
case normal
case goingAway
case protocolError
case tlsHandshake
case raw(UInt16)
```

**Comment by houndci-bot-dead** [comment]
I changed my mind, your code is awesome!

**Comment by Danappelxx** [comment]
Haha what the hell?

**Comment by marchinram** [comment]
I guess it can change its mind

**Comment by paulofaria** [comment]
what's left to merge this beauty?

**Comment by paulofaria** [comment]
besides the conflict?

**Comment by marchinram** [comment]
Sorry been busy at work, gonna have time this weekend, I was having some merge issues, also I still haven't been able to test the client out, I followed advice from @tomohisa but still can't don't want to merge until client passes all tests too

**Comment by vi4m** [comment]
Fingers crossed 👍

**Comment by paulofaria** [comment]
@marchinram can you check what's making the tests fail and merge? If you don't have time I can do it later at night. (:

**Comment by antonmes** [comment]
screen shot 2016-05-31 at 20 43 41

**Comment by antonmes** [comment]
@tomohisa test case is 100% correct, right?

**Comment by paulofaria** [comment]
how is it passing autobahn if that's wrong? @marchinram

**Comment by antonmes** [comment]
I'm not sure this test is correct. If it doesn't change the string then the `frame.data` is correct, because
'Hello' equals [0x48, 0x65, 0x6c, 0x6c, 0x6f] in byte representation.

**Comment by tomohisa** [comment]
@antonmes this test should be correct. when you send data from client, you need to mask data with `maskKey`. That's why this work with server, but not with client.
https://github.com/Zewo/WebSocket/blob/master/Sources/Frame.swift#L137
check this from original one.
before I change this, it was not working connecting with ws://echo.websocket.org but after this change, it works finally fine with this.

**Comment by tomohisa** [comment]
https://tools.ietf.org/html/rfc6455#section-5.3
server never mask data but client mask data.

**Comment by marchinram** [comment]
I'm gonna check this out later tonight not sure why this is failing but wasn't before

**Comment by tomohisa** [comment]
@marchinram if you see original code, you should do same masking, so it shouldn't be that hard.
https://github.com/Zewo/WebSocket/blob/master/Sources/Frame.swift#L137

**Comment by Joannis** [comment]
@tomohisa is right. Servers don't mask, clients do.

**Comment by marchinram** [comment]
I pushed changes that fixes travis, @antonmes i literally just saw your pr in my fork after pushing this sorry bout that can include your version instead if you want

**Comment by antonmes** [comment]
@marchinram no, it's ok :) let's merge this beauty

**Comment by marchinram** [comment]
@antonmes cool ok @paulofaria can you merge don't have permissions, also i'm freed up more now if we wanna split this up to client and server implementation we should all discuss how we want to get this done

**Comment by paulofaria** [comment]
@marchinram I think it would be `WebSocket` with `Socket.swift`, `Frame.swift` and `CloseCode.swift` and `WebSocketServer` and `WebSocketClient`. Simple enough, right?

**Comment by paulofaria** [comment]
@marchinram did the API change?

**Comment by marchinram** [comment]
@paulofaria I don't believe any public facing apis have changed, is some test or something failing?

**Comment by antonmes** [comment]
@marchinram we don't have integration tests, but we can bump minor just in case, why not? :)

**Comment by marchinram** [comment]
@antonmes yeah agreed it was a decent sized change can probably bump it

**Comment by paulofaria** [comment]
why do we need `UUID`?

**Comment by marchinram** [comment]
Don't think we do its not in the package dependencies afaik

**Comment by paulofaria** [comment]
https://github.com/Zewo/WebSocket/blob/master/Sources/Socket.swift#L370

It's to make the socket `Hashable`. @marchinram any idea about why it needs to be `Hashable`? I think someone was trying to put the sockets as the key of a `Dictionary` or in a `Set`. Something like that. Wasn't it you? haha

**Comment by paulofaria** [comment]
@marchinram yeah it was you.. haha. If you don't mind, I'll remove that dependency, is that cool? You can add it in your project in an extension if you need it. `UUID` needs an extra step installing it as a system package, so it affects people that don't want to use it in a `Dictionary` key or in a `Set`, etc.

**Pull request merged**